game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsFactory.java

```diff
@@ -0,0 +1,53 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.DiceRoll;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.NonNull;
+
+/** Build the steps for the fire round (roll dice, select casualties, and mark casualties) */
+@Builder
+public class FireRoundStepsFactory {
+
+  @NonNull final BattleState battleState;
+  @NonNull final BattleActions battleActions;
+  @NonNull final Function<BattleState, List<FiringGroup>> firingGroupSplitter;
```

Comment:
What is the implied ordering of the output list from firingGroupSplitter? What contract would instances of this local variable need to satisfy to be consistent with the expected usages within this class?

Reply:
I'll change it to `Collection`.  This class doesn't care about the order.